cpp-ch/local-engine/local_engine_jni.cpp

```diff
@@ -1039,7 +1039,7 @@ JNIEXPORT jobject Java_org_apache_spark_sql_execution_datasources_CHDatasourceJn
     local_engine::BlockStripes bs = local_engine::BlockStripeSplitter::split(*block, partition_col_indice_vec, hasBucket, reserve_);
 
     auto * addresses = env->NewLongArray(bs.block_addresses.size());
-    env->SetLongArrayRegion(addresses, 0, bs.block_addresses.size(), bs.block_addresses.data());
+    env->SetLongArrayRegion(addresses, 0, bs.block_addresses.size(), (const jlong *)bs.block_addresses.data());
```

Comment:
using static_cast

Reply:
may be reinterpret_cast

Reply:
> may be reinterpret_cast

let's use `static_cast` first for safety